homeassistant/components/recorder/__init__.py

```diff
@@ -214,6 +215,61 @@
 # Pool size must accommodate Recorder thread + All db executors
 MAX_DB_EXECUTOR_WORKERS = POOL_SIZE - 1
 
+pool_lock = threading.RLock()
```

Comment:
This makes test affect each other if the lock is not released when a test ends.
Initializing the lock should be done when setting up the recorder instead.